test_data/pil/fib_macro.pil

```diff
@@ -20,16 +20,13 @@ pol constant ISLAST(i) { one_hot(i, %last_row) };
 
 // ANCHOR: constraint_macro_definitions
 macro constrain_equal_expr(A, B) { A - B };
-macro force_equal_on_last_row(poly, value) { ISLAST * constrain_equal_expr(poly, value) = 0; };
+macro force_equal_on_first_row(poly, value) { ISLAST * constrain_equal_expr(poly', value) = 0; };
```

Comment:
I'm still confused by whether we're doing cyclic constraints or not. Are we?

Reply:
I see no reason why we shouldn't allow recursive function calls. In the end of course, after inlining all functions, the resulting constraint has to be finite. But I don't see a strong argument against:
```
let first_n_zeros = |X, n| match i {
  0 => X,
  n => (X - n) * first_n_zeros(n - 1),
}
let wit_col;
first_n_zeros(wit_col, 3) = 0;
```
which would result in 
```
(wit_col - 2) * (wit_col - 1) * wit_col = 0
```

Reply:
that's not what I mean, I mean next(last) = first

Reply:
I'm always assuming the rows are cyclic.

Reply:
ok